Documentation/upgrading.md

```diff
@@ -1,173 +1,143 @@
+# Upgrading self-hosted Kubernetes
 
-# Upgrading Self-hosted Kubernetes
+"Self-hosted" Kubernetes clusters run the apiserver, scheduler, controller-manager, kube-dns, kube-proxy, and flannel or calico as pods, like ordinary applications (except with taint tolerations). This allows upgrades to be performed in-place using (mostly) `kubectl` as an alternative to re-provisioning.
 
-Self-hosted Kubernetes clusters schedule Kubernetes components such as the apiserver, kubelet, scheduler, and controller-manager as pods like other applications (except with node selectors). This allows Kubernetes level operations to be performed to upgrade clusters in place, rather than by re-provisioning.
+Let's upgrade a Kubernetes v1.6.6 cluster to v1.6.7 as an example.
 
-Let's upgrade a self-hosted Kubernetes v1.4.1 cluster to v1.4.3 as an example.
+## Status
 
-## Inspect
+bootkube does not provide in-place patch diffs or migrations between releases. In-place upgrades are inherently more dangerous than a blue/green cluster deployments. Evaluate whether an in-place upgrade is appropriate for your Kubernetes cluster and risk tolerance before beginning. Failover important workloads to another cluster in advance.
```

Comment:
I feel like this might be a little unnecessarily scary. In most cases, even if the control plane is down the workloads are still running fine. Rather than this, I feel like we should have a section on rollbacks incase of failure scenarios (difficult to assume, enumerate).

Reply:
Any specific wording changes you'd suggest? I agree with the desire to have more robust docs about how to recover from each possible sticky situation. I don't have those docs.

I really want folks to weight whether the risks of in-place upgrades are tolerable for their cluster, since bootkube itself isn't providing guarantees if something goes wrong.

To an admin responsible for keeping production services running, creating a new cluster, deploying the same workloads, verifying services work properly, migrating traffic, and later (once you're sure) tearing down the old cluster is the responsible way to go. At least from the "don't get fired" perspective. 

For other situations, editing the control plane may be perfectly tolerable, especially if its a simple diff. Still, it is more dangerous. Its easy to typo something and end up with an inaccessible control plane. Even if "workloads are running fine", the cluster is not properly tolerating node failures, shifting pods, adding new workloads, etc.